librz/core/cmd/cmd_analysis.c

```diff
@@ -3095,143 +2994,122 @@ RZ_API void rz_cmd_analysis_calls(RzCore *core, const char *input, bool printCom
 	RzListIter *iter;
-	RzList *list;
-	RzNum *num = NULL;
-	int n;
+	RzSyscallItem *si;
+	RzList *list = rz_syscall_list(core->analysis->syscall);
+	rz_cmd_state_output_array_start(state);
+	rz_list_foreach (list, iter, si) {
+		switch (state->mode) {
+		case RZ_OUTPUT_MODE_STANDARD:
+			rz_cons_printf("%s = 0x%02x.%s\n",
+				si->name, si->swi, syscallNumber(si->num));
+			break;
+		case RZ_OUTPUT_MODE_JSON:
+			pj_o(state->d.pj);
+			pj_ks(state->d.pj, "name", si->name);
+			pj_ki(state->d.pj, "swi", si->swi);
+			pj_ki(state->d.pj, "num", si->num);
+			pj_end(state->d.pj);
+			break;
+		default:
+			rz_warn_if_reached();
+			break;
+		}
+	}
+	rz_cmd_state_output_array_end(state);
+	rz_list_free(list);
+	return RZ_CMD_STATUS_OK;
+}
 
-	switch (input[0]) {
-	case 'c': // "asc"
-		if (input[1] == 'a') {
-			if (input[2] == ' ') {
-				if (!isalpha((ut8)input[3]) && (n = rz_num_math(num, input + 3)) >= 0) {
-					si = rz_syscall_get(core->analysis->syscall, n, -1);
-					if (si) {
-						rz_cons_printf(".equ SYS_%s %s\n", si->name, syscallNumber(n));
-						rz_syscall_item_free(si);
-					} else
-						eprintf("Unknown syscall number\n");
-				} else {
-					n = rz_syscall_get_num(core->analysis->syscall, input + 3);
-					if (n != -1) {
-						rz_cons_printf(".equ SYS_%s %s\n", input + 3, syscallNumber(n));
-					} else {
-						eprintf("Unknown syscall name\n");
-					}
-				}
-			} else {
-				list = rz_syscall_list(core->analysis->syscall);
-				rz_list_foreach (list, iter, si) {
-					rz_cons_printf(".equ SYS_%s %s\n",
-						si->name, syscallNumber(si->num));
-				}
-				rz_list_free(list);
-			}
-		} else {
-			if (input[1] == ' ') {
-				if (!isalpha((ut8)input[2]) && (n = rz_num_math(num, input + 2)) >= 0) {
-					si = rz_syscall_get(core->analysis->syscall, n, -1);
-					if (si) {
-						rz_cons_printf("#define SYS_%s %s\n", si->name, syscallNumber(n));
-						rz_syscall_item_free(si);
-					} else
-						eprintf("Unknown syscall number\n");
-				} else {
-					n = rz_syscall_get_num(core->analysis->syscall, input + 2);
-					if (n != -1) {
-						rz_cons_printf("#define SYS_%s %s\n", input + 2, syscallNumber(n));
-					} else {
-						eprintf("Unknown syscall name\n");
-					}
-				}
-			} else {
-				list = rz_syscall_list(core->analysis->syscall);
-				rz_list_foreach (list, iter, si) {
-					rz_cons_printf("#define SYS_%s %s\n",
-						si->name, syscallNumber(si->num));
-				}
-				rz_list_free(list);
+RZ_IPI RzCmdStatus rz_analysis_syscall_name_handler(RzCore *core, int argc, const char **argv) {
+	st64 num = rz_syscall_get_num(core->analysis->syscall, argv[1]);
+	if (num < 1) {
+		RZ_LOG_ERROR("Cannot resolve syscall: %s\n", argv[1]);
+		return RZ_CMD_STATUS_ERROR;
+	}
+	rz_cons_println(syscallNumber(num));
+	return RZ_CMD_STATUS_OK;
+}
+
+RZ_IPI RzCmdStatus rz_analysis_syscall_number_handler(RzCore *core, int argc, const char **argv) {
+	st64 num = rz_num_math(core->num, argv[1]);
```

Comment:
AFAIR we should use `rz_num_math(NULL, ...)` when we don't want resolution of flags, function names, etc. like in this case (it is a _NUM argument, not a _RZNUM).